ostd/src/mm/page_table/node.rs

```diff
@@ -231,18 +231,18 @@ where
     /// set the lock bit for performance as it is exclusive and unlocking is an
     /// extra unnecessary expensive operation.
     pub(super) fn alloc(level: PagingLevel) -> Self {
-        let page = page::allocator::PAGE_ALLOCATOR.get().unwrap().lock().alloc_page(PAGE_SIZE).map(|paddr|{
-            Page::<PageTablePageMeta<E, C>>::from_unused(paddr)
-        }).unwrap_or_else(||{
-            panic!("Failed to allocate a page for a page table node")
-        });
-
-        // The lock is initialized as held.
-        page.meta().lock.store(1, Ordering::Relaxed);
-
-        // SAFETY: here the page exclusively owned by the newly created handle.
-        let inner = unsafe { &mut *page.meta().inner.get() };
-        inner.level = level;
+        let page = page::allocator::PAGE_ALLOCATOR
+            .get()
+            .unwrap()
+            .lock()
+            .alloc_page(PAGE_SIZE)
+            .map(|paddr| {
+                Page::<PageTablePageMeta<E, C>>::from_unused(
+                    paddr,
+                    PageTablePageMeta::new_locked(level),
+                )
+            })
+            .unwrap_or_else(|| panic!("Failed to allocate a page for a page table node"));
```

Comment:
Don't do `.unwrap_or_else(|| panic!(`. Also, `from_unused` shouldn't be scattered all over the place.